rclpy/rclpy/task.py

```diff
@@ -78,15 +84,15 @@ def cancelled(self) -> bool:
 
         :return: True if the task was cancelled
         """
-        return self._cancelled
+        return self._state == FutureState.CANCELLED
 
     def done(self) -> bool:
         """
         Indicate if the task has finished executing.
 
         :return: True if the task is finished or raised while it was executing
         """
-        return self._done
+        return self._state != FutureState.PENDING
```

Comment:
@nadavelkabets 

To keep the consistent behavior including backports for jazzy and humble, probably we want to keep this as following?

```suggestion
        return self._state == FutureState.FINISHED
```